### PR TITLE
Jhalackova vmhostd

### PR DESCRIPTION
**Pull request by lvicoun**
### Description

Added vhostmd description.

### Are there any relevant issues/feature requests?

* https://jira.suse.com/browse/SMO-397

### Is this (based on) existing content?

* old xml:id...
* old URL...

### CODE REVIEW

**Comment by lvicoun** [comment]
[Micro-vhostmd_en.pdf](https://github.com/user-attachments/files/16994268/Micro-vhostmd_en.pdf)

File: articles/vhostmd.asm.xml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+              and can provide those metrics to virtual machines running on that host.
+            </para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>WHY?</term>
+          <listitem>
+            <para>
+              The article describes how to install and configure <literal>vhostmd</literal> to
+              enable you collecting metrics.
```
Comment (jfehlig) [inline articles/vhostmd.asm.xml:147]: This sentence sounds a little awkward. I'd suggest removing 'you'. E.g. "... to enable collecting metrics."

File: concepts/vhostmd_about.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    <abstract><!-- can be changed via merge in the assembly -->
+      <para>
+       
+      </para>
+    </abstract>
+  </info>
+  <para>
+<literal>vhostdmd</literal> is a daemon that runs in a host environment and provides a metrics
+communication channel between the host and guest machines. The daemon provides a limited set of
+host resources consumption to the guest machine administrator (susually in a read-only mode), to help analyse problems with the
```
Comment (jfehlig) [inline concepts/vhostmd_about.xml:33]: s/susually/usually/

File: concepts/vhostmd_about.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+guest machine.
+</para>
+<para>
+  <literal>vhostdmd</literal> writes metrics periodically to a disk. What is to record, the
```
Comment (jfehlig) [inline concepts/vhostmd_about.xml:37]: The second sentence sounds a little awkward to me. Maybe rephrase it to something like "What metrics are collected, the frequency which they are collected, and where they are written can be controlled by a configuration file."

File: tasks/vhostmd_providing_information.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    <!-- add author's e-mail -->
+    <meta name="maintainer" content="jsindelarova@suse.com" its:translate="no"/>
+    <abstract><!-- can be changed via merge in the assembly -->
+      <para>
+       The topic covers steps to make collected metrics visible to a virtual machine.
+      </para>
+    </abstract>
+  </info>
+  <para>
+    You can make the collected metrics vsible to particular virtual machines for further usage or
```
Comment (jfehlig) [inline tasks/vhostmd_providing_information.xml:32]: s/vsible/visible/

File: tasks/vhostmd_providing_information.xml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    just for reading. There are two ways to achieve this. Either you can atach the metrics disk using the &virsh; command or
```
Comment (jfehlig) [inline tasks/vhostmd_providing_information.xml:33]: s/atach/attach/

File: tasks/vhostmd_providing_information.xml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    </para>
+    <screen>
+&lt;disk type='block' device='disk'>
+  &lt;source dev='/dev/shm/vhostmd0'/>
+  &lt;target dev='hdb' bus='ide'/>
+  &lt;readonly/>
+&lt;/disk>
+    </screen>
+    <para>
+      Adjust the <literal>dev</literal> and <literal>dbus</literal> valuse of the target.
```
Comment (jfehlig) [inline tasks/vhostmd_providing_information.xml:66]: s/valuse/values/

File: references/vhostmd_configuration.xml [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    <meta name="maintainer" content="jsindelarova@suse.com" its:translate="no"/>
+    <abstract><!-- can be changed via merge in the assembly -->
+      <para>
+       Introductory text
+      </para>
+    </abstract>
+  </info>
+  <para>
+    The default configuration file of <literal>vhostmd</literal> is located in
+    <filename>/etc/vhostmd/vhostmd.conf</filename>. Yoy can adjust this files to suit yor needs. The configuration file is an XML file that must
```
Comment (jfehlig) [inline references/vhostmd_configuration.xml:32]: s/Yoy/You and s/yor/your/

File: references/vhostmd_configuration.xml [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+            </para>
+          </listitem>
+        </itemizedlist>
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term><literal>&lt;update_period&gt;</literal></term>
+      <listitem>
+        <para>
+          Defines how often are the metric updated. The value is in seconds.
```
Comment (jfehlig) [inline references/vhostmd_configuration.xml:79]: Suggest: "Defines how often the metrics are updated."

**Review by jfehlig — COMMENTED** [review]
FYI, the SAP team has also written some instructions on using vhostmd
https://confluence.suse.com/display/SAP/HANAonKVM+vhostmd

File: concepts/vhostmd_data_format.xml [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+       The topic covers details about the data format of metrics and also how the output in XML
+       looks like.
+      </para>
+    </abstract>
+  </info>
+  <para>
+    <literal>vhostmd</literal> stores the collected metrics data on the metrics disk in a raw
+    format. The disk contains 32bytes header followed by the metrics data in XML format.
+  </para>
+  <section xml:id="vhostmd-data-format-xml">
```
Comment (dariavladykina) [inline concepts/vhostmd_data_format.xml:35]: This is an orphaned section. Maybe consider restructuring?

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some suggestions from me. Thanks!

File: references/vhostmd_configuration.xml [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+          Specify how the metrics are transported to guest machines. In case of &productname;, use
+          the default of <literal>virtio</literal>.
+        </para>
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term><literal>&lt;virtio&gt;</literal></term>
+      <listitem>
+        <para>
+Here you can use the following elements:
```
[the remark this answers is not in the document]
Comment (lvicoun) [reply references/vhostmd_configuration.xml:96]: Not here, as those are elements. Attributes are stated here as well and they modify the element behaviour.
